packages/react/src/experimental/OneDataCollection/index.tsx

```diff
@@ -440,20 +487,36 @@ export const OneDataCollection = <
           </div>
           <Filters.ChipsList />
         </Filters.Root>
-        <VisualizationRenderer
-          visualization={visualizations[currentVisualization]}
-          source={source}
-          onSelectItems={onSelectItemsLocal}
-          onTotalItemsChange={setTotalItems}
-        />
-        {bulkActions?.primary && (bulkActions?.primary || []).length > 0 && (
-          <OneActionBar
-            isOpen={showActionBar}
-            selectedNumber={selectedItemsCount}
-            primaryActions={bulkActions.primary}
-            secondaryActions={bulkActions?.secondary}
-            onUnselect={() => clearSelectedItemsFunc?.()}
-          />
+
+        {emptyState ? (
+          <div className="flex flex-col items-center justify-center">
+            <OneEmptyState
+              emoji={emptyState.emoji}
+              title={emptyState.title}
+              description={emptyState.description}
+              actions={emptyState.actions}
+            />
+          </div>
+        ) : (
+          <>
+            <VisualizationRenderer
+              visualization={visualizations[currentVisualization]}
+              source={source}
+              onSelectItems={onSelectItemsLocal}
+              onLoadData={onLoadData}
+              onLoadError={onLoadError}
+            />
+            {bulkActions?.primary &&
+              (bulkActions?.primary || []).length > 0 && (
+                <OneActionBar
+                  isOpen={showActionBar}
+                  selectedNumber={selectedItemsCount}
+                  primaryActions={bulkActions.primary}
+                  secondaryActions={bulkActions?.secondary}
+                  onUnselect={() => clearSelectedItemsFunc?.()}
+                />
+              )}
```

Comment:
Would you consider to set the condition in a const:
`const hasPrimaryBulkActions = Array.isArray(bulkActions?.primary) && bulkActions.primary.length > 0;`
```
{hasPrimaryBulkActions && (
  <OneActionBar
    isOpen={showActionBar}
    selectedNumber={selectedItemsCount}
    primaryActions={bulkActions.primary}
    secondaryActions={bulkActions?.secondary}
    onUnselect={clearSelectedItemsFunc}
  />
)}
```